intf/intf_test.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intf
+
+import (
+	"fmt"
+	"net"
+	"testing"
+)
+
+type validInterfaceAccessor struct {
+	unimplementedAccessor
+}
+
+func (validInterfaceAccessor) Interface(name string) (*Interface, error) {
+	switch name {
+	case "eth0":
+		return &Interface{}, nil
+	case "eth42":
+		return &Interface{}, fmt.Errorf("error")
+	}
+	return nil, fmt.Errorf("does not exist")
+}
+
+func TestValidInterface(t *testing.T) {
+	oldAccessor := accessor
+	defer func() {
+		accessor = oldAccessor
+	}()
+	accessor = validInterfaceAccessor{}
+
+	tests := []struct {
+		name string
+		in   string
+		want bool
+	}{{
+		name: "exists",
+		in:   "eth0",
+		want: true,
+	}, {
+		name: "does not exist",
+		in:   "eth42",
+		want: false,
+	}}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got, want := ValidInterface(tt.in), tt.want; got != want {
+				t.Fatalf("InterfaceValid(%s), did not get expected interface valid response, got: %v, want: %v", tt.in, got, want)
+			}
+		})
+	}
+}
+
+type addIPAccessor struct {
+	unimplementedAccessor
```

Comment:
I do see the use of this unimplementedAccessor for the testing structure - so you can inherit everything to be the unimplemented versions by default and then override whatever one you need